apps/nextjs/src/app/[locale]/auth/login/_login-form.tsx

```diff
@@ -14,65 +15,138 @@ import { validation } from "@homarr/validation";
-        router.push("/");
+        router.push(callbackUrl);
+      }
+    },
+    [t, router, callbackUrl],
+  );
+
+  const onError = useCallback(() => {
+    setIsPending(false);
+
+    showErrorNotification({
+      title: t("action.login.notification.error.title"),
+      message: t("action.login.notification.error.message"),
+      autoClose: 10000,
+    });
+  }, [t]);
+
+  const signInAsync = useCallback(
+    async (provider: string, options?: Parameters<typeof signIn>[1]) => {
+      setIsPending(true);
+      await signIn(provider, {
+        ...options,
+        redirect: false,
+        callbackUrl: new URL(callbackUrl, window.location.href).href,
       })
-      .catch((error: Error | string) => {
-        setIsLoading(false);
-        setError(error.toString());
-        showErrorNotification({
-          title: t("action.login.notification.error.title"),
-          message: t("action.login.notification.error.message"),
-        });
-      });
-  };
+        .then(onSuccess)
+        .catch(onError);
+    },
+    [setIsPending, onSuccess, onError, callbackUrl],
+  );
+
+  const isLoginInProgress = useRef(false);
+
+  useEffect(() => {
+    if (isOidcAutoLoginEnabled && !isPending && !isLoginInProgress.current) {
+      isLoginInProgress.current = true;
+      void signInAsync("oidc");
+    }
+  }, [signInAsync, isOidcAutoLoginEnabled, isPending]);
 
   return (
     <Stack gap="xl">
-      <form onSubmit={form.onSubmit((values) => void handleSubmitAsync(values))}>
-        <Stack gap="lg">
-          <TextInput label={t("field.username.label")} {...form.getInputProps("name")} />
-          <PasswordInput label={t("field.password.label")} {...form.getInputProps("password")} />
-          <Button type="submit" fullWidth loading={isLoading}>
-            {t("action.login.label")}
+      <Stack gap="lg">
+        {credentialInputsVisible && (
+          <>
+            <form onSubmit={form.onSubmit((credentials) => void signInAsync("credentials", credentials))}>
+              <Stack gap="lg">
+                <TextInput label={t("field.username.label")} {...form.getInputProps("name")} />
+                <PasswordInput label={t("field.password.label")} {...form.getInputProps("password")} />
+
+                {providers.includes("credentials") && (
+                  <SubmitButton isPending={isPending} form={form} credentialType="basic">
+                    {t("action.login.label")}
+                  </SubmitButton>
+                )}
+
+                {providers.includes("ldap") && (
+                  <SubmitButton isPending={isPending} form={form} credentialType="ldap">
+                    {t("action.login.labelWith", { provider: "LDAP" })}
+                  </SubmitButton>
+                )}
+              </Stack>
+            </form>
+            {providers.includes("oidc") && <Divider label="OIDC" labelPosition="center" />}
+          </>
+        )}
+
+        {providers.includes("oidc") && (
```

Comment:
You can add a text to say they can click the button if they don't get redirected automatically.
(only when isOidcAutoLoginEnabled is true)

Reply:
I think it should be clear enough to the user that he can click that button and the text seems unnecessary to me